common/http/src/retry.rs

```diff
@@ -132,9 +134,22 @@ impl Retry {
     }
 
     async fn read_from_disk(&self, path: &Path) -> Result<(Option<OffsetMap>, IngestBody), Error> {
-        let mut file = BufReader::new(File::open(path).await?);
         let mut data = String::new();
-        file.read_to_string(&mut data).await?;
+        {
```

Comment:
Can we remove this block around the code? The only reason I could see for it would be ensuring that the buffers and decoder are deallocated before the remainder of the function is called and it doesn't seem to be that much need for the slightly earlier deallocation.

It just looks odd and not like idomatic rust code that I've seen.

Reply:
It is needed to grantee that file is closed. Otherwise size calcs and file stats later in the same function may be affected.

Reply:
I could easily see someone removing that block in the future and commit it because all the CI tests pass. So we should either build a test that will fail when changed or comment it to communicate the intent. We've never seen a failure from any of our tests that are checking that files are cleaned up so it would be easy for someone to undo that without understanding the intent.

Reply:
added comments to scope blocks.